crates/matrix-sdk-crypto/src/identities/user.rs

```diff
@@ -291,6 +292,36 @@ impl UserIdentity {
             methods,
         )
     }
+
+    /// Pin the current identity (Master public key).
+    pub async fn pin_current_master_key(&self) -> Result<(), CryptoStoreError> {
+        self.inner.pin();
+        let to_save = ReadOnlyUserIdentities::Other(self.inner.clone());
+        let changes = Changes {
+            identities: IdentityChanges { changed: vec![to_save], ..Default::default() },
+            ..Default::default()
+        };
+        self.verification_machine.store.inner().save_changes(changes).await?;
+        Ok(())
+    }
+
+    /// An identity mismatch is detected when there is a trust problem with the
+    /// user identity. There is an identity mismatch if the current identity
+    /// is not verified and there is a pinning violation. An identity
+    /// mismatch must be reported to the user, and can be resolved by:
+    /// - Verifying the new identity (see
+    ///   [`UserIdentity::request_verification`])
+    /// - Or by updating the pinned key
+    ///   ([`UserIdentity::pin_current_master_key`]).
+    pub fn has_identity_mismatch(&self) -> bool {
```

Comment:
Similar to the `has_pin_violation()` method, the first line should be something akin to "Has the identity changed since we last pinned it?"

I think this could also could be perhaps better named, identity mismatch makes me think of the identity not being what we expect. I guess you could argue that this is exactly the case here, but perhaps we could name it to be something along the lines that the identity has changed without our approval instead. Feel free to ignore this part if you disagree.

Reply:
I am not sure what would be the best name for that. I took identity mismatch to use the same term that is used in this [doc](https://github.com/element-hq/element-meta/blob/develop/docs/crypto/trust_v2.md#behavior-for-identity-mismatch).

Maybe @dkasak have an idea? Maybe `is_identity_trustworthy`? or more based on the use of this method `identity_needs_user_approval` / `validation` / `review` ?

Reply:
I would avoid the `mismatch` and `trustworthy` wording. `identity_needs_user_approval` sounds good to me.